WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,334 @@
+    // They're lazy because of `self` implementing `alertsPresenter`.
+    // If that were done by another owned struct, and the handler passed to it as and when it changed, we could do this in the init.
+    private lazy var bluetoothConnectionController: CardReaderConnectionController = {
+        CardReaderConnectionController(
+            forSiteID: siteID,
+            knownReaderProvider: CardReaderSettingsKnownReaderStorage(),
+            alertsPresenter: self,
+            alertsProvider: BluetoothReaderConnectionAlertsProvider(),
+            configuration: CardPresentConfigurationLoader().configuration,
+            analyticsTracker: CardReaderConnectionAnalyticsTracker(
+                configuration: CardPresentConfigurationLoader().configuration,
+                siteID: siteID,
+                connectionType: .userInitiated, // This will need a tweak – it changes from use to use.
+                analytics: ServiceLocator.analytics))
+    }()
+
+    private lazy var tapToPayConnectionController: BuiltInCardReaderConnectionController = {
+        BuiltInCardReaderConnectionController(
+            forSiteID: siteID,
+            alertsPresenter: self,
+            alertsProvider: BluetoothReaderConnectionAlertsProvider(),
+            configuration: CardPresentConfigurationLoader().configuration,
+            analyticsTracker: CardReaderConnectionAnalyticsTracker(
+                configuration: CardPresentConfigurationLoader().configuration,
+                siteID: siteID,
+                connectionType: .userInitiated, // This will need a tweak – it changes from use to use.
+                analytics: ServiceLocator.analytics))
+    }()
+
+    private let onboardingPresenterAdaptor: CardPresentPaymentsOnboardingPresenterAdaptor
+
+    private var paymentScreenEventSubject = CurrentValueSubject<CardPresentPaymentEvent?, Never>(nil)
+
+    required init(siteID: Int64) {
+        self.siteID = siteID
+        onboardingPresenterAdaptor = CardPresentPaymentsOnboardingPresenterAdaptor(stores: ServiceLocator.stores)
+
+        paymentScreenEventPublisher = onboardingPresenterAdaptor.onboardingScreenViewModelPublisher.map { onboardingViewModel -> CardPresentPaymentEvent? in
+            guard let onboardingViewModel else {
+                return nil
+            }
+            return CardPresentPaymentEvent.showOnboarding(onboardingViewModel)
+        }
+        .merge(with: paymentScreenEventSubject)
+        .eraseToAnyPublisher()
+    }
+
+    private var paymentTask: Task<CardPresentPaymentResult, Never>?
+
+    @MainActor
+    func collectPayment(for order: Order,
+                        using discoveryMethod: CardReaderDiscoveryMethod) async -> CardPresentPaymentResult {
+        let preflightController = CardPresentPaymentPreflightController(
+            siteID: siteID,
+            configuration: CardPresentConfigurationLoader().configuration,
+            rootViewController: UIViewController(),
```

Comment:
Yep, a `NullPresenting` instance would definitely be better here. 

I can't remember whether I missed this, or deliberately left it because it needed something from `UIViewController` which was harder to articulate in the Presenting protocol.